core/arch/arm/plat-rockchip/platform_rk3588.c

```diff
@@ -389,6 +416,49 @@ TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 	/* Clear buffer memory */
 	memzero_explicit(buffer, sizeof(buffer));
 
+	return res;
+}
+
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	TEE_Result res = TEE_SUCCESS;
+
+	if (!hwkey)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	mutex_lock(&huk_mutex);
+
+	/* Try to use the cached HUK from memory */
+	if (huk)
+		goto out;
+
+	huk = malloc(sizeof(*huk));
+	if (!huk) {
+		res = TEE_ERROR_OUT_OF_MEMORY;
+		goto out;
+	}
+
+	/* Try to read and cache the HUK persisted in the OTP */
+	res = read_huk(huk);
+	if (res != TEE_ERROR_NO_DATA)
+		goto out;
+
+	/* Try to generate and use a new HUK and persist it in the OTP */
+	res = generate_huk(huk);
+	if (res != TEE_SUCCESS)
+		goto out;
+	res = persist_huk(huk);
+
+out:
+	if (res == TEE_SUCCESS) {
+		memcpy(hwkey->data, huk->data, HW_UNIQUE_KEY_LENGTH);
+	} else if (huk) {
+		/* Get rid of cached HUK in case of an error */
+		memzero_explicit(huk, sizeof(*huk));
+		free(huk);
```

Comment:
Suggestion (fell free to address or not):
```suggestion
	} else {
		free_wipe(huk);